contracts/javascore/Intent_Contracts/app/src/main/java/network/icon/intent/Intent.java

```diff
@@ -0,0 +1,348 @@
+
+        OrderCancelled(order.id, order.srcNID);
+    }
+
+    void _resolveFill(String srcNetwork, OrderFill _fill) {
+        SwapOrder order = orders.get(_fill.id);
+        if (order == null) {
+            Context.revert("There is no order to resolve");
+        }
+        Context.require(Arrays.equals(Context.hash("keccak-256", order.toBytes()),
+                Context.hash("keccak-256",
+                        _fill.orderBytes)),
+                "Mismatched order");
+
+        Context.require(order.dstNID.equals(srcNetwork), "Invalid Network");
+
+        orders.set(_fill.id, null);
+        OrderClosed(_fill.id);
+
+        Address tokenAddress = Address.fromString(extractAddress(order.token));
+        if (tokenAddress.equals(nativeAddress.get())) {
+            Context.transfer(Address.fromString(extractAddress(_fill.solver)), order.amount);
+        } else {
+            Context.call(tokenAddress, "transfer", Address.fromString(extractAddress(_fill.solver)), order.amount);
+        }
+    }
+
+    void _transferResult(String _toAddress, String _toToken, BigInteger amount,
+            BigInteger fee) {
+        Address toAddress = Address.fromString(extractAddress(_toAddress));
+        Address toTokenAddress = Address.fromString(extractAddress(_toToken));
+        if (toTokenAddress.equals(nativeAddress.get())) {
+            Context.require(Context.getValue().equals(amount.add(fee)), "\"Deposit amount not equal to order amount\"");
+            _nativeTransfer(toAddress, amount);
+            _nativeTransfer(feeHandler.get(), fee);
+        } else {
+            Context.call(toTokenAddress, "transfer", toAddress,
+                    amount);
+            Context.call(toTokenAddress, "transfer",
+                    feeHandler.get(), fee);
+        }
+    }
+
+    void _nativeTransfer(Address to, BigInteger amount) {
+        Context.transfer(to, amount);
+    }
+
+    @External
+    public void tokenFallback(Address _from, BigInteger _value, byte[] _data) {
+        Context.require(_value.compareTo(BigInteger.ZERO) > 0, "Zero transfers not allowed");
+
+        String unpackedData = new String(_data);
+        Context.require(!unpackedData.equals(""), "Token Fallback: Data can't be empty");
+        JsonObject json = Json.parse(unpackedData).asObject();
+
+        // string(address of depositer) -> string (deposits token address) -> amountof
+        // token
+        // String depositor = json.get("depositor").asString();
+        // String token = json.get("token").asString();
+        // BigInteger amount = new BigInteger(json.get("amount").asString());
+        String type = json.get("type").asString();
+        // deposit.at(depositor).set(token, amount);
+        if (type.equals("swap")) {
+            SwapOrder swapOrder = SwapOrder
+                    .fromBytes(hexStringToByteArray(json.get("swapOrderDataBytes").asString()));
+            deposit.at(swapOrder.creator).set(swapOrder.token, swapOrder.amount);
```

Comment:
Should not be needed should just be able to verify the token and amount matches the swap order here